### DISCUSSION TITLE
Presenting data from a cloud database onto an Outline page?

### DISCUSSION THREAD

**Discussion by RickCogley** [outline/outline#1744] [Self Hosting]
We have an ops database in which we put our time & expense, contacts, projects, leads and so on. This database has an API, and a way to either a) pull json or csv of a given view (query result), or b), embed a javascript that presents a view. Regarding a) I can use js to pull data and present it and regarding b) I can simply paste a <script></script> into an html page to get a list. So, there’s some flexibility there.

Can anyone suggest a way to present our db’s info on an (hosted cloud) Outline page?

**Comment by tommoor** [comment]
Outline does not support embedding script tags for security, however there are a couple of options:

- Using a small script and the [documents.update](https://www.getoutline.com/developers#tag/Documents/paths/~1documents.update/post) endpoint – write a markdown formatted table or other formatted output to the document of your choice on a regular basis.
- Perhaps the output of the html page has a remotely hosted image that could be hotlinked from Outline? You can use Markdown syntax again to insert an image that is hosted on a remote server (your ops database?)
- Is the ops database another hosted product? Perhaps we could persue a more official integration for other services.

**Reply by RickCogley** [reply]
Right, right, Outline has an API. That would work for us. Thanks! 

So, the only thing that is keeping us from subscribing is image formatting. We have markdown docs in our ops db, but the reason we are trying to move away from it is its image processing. It’s got a lot of “friction” and so ppl are reluctant, even creating rogue documents in (gasp!) MS Word! 

Even a selector allowing say, three sizes would be great. Then ppl can paste a screenshot and really quickly size it as appropriate.

**Reply by matbrgz** [reply]
@RickCogley Could you provide more information about what you want to do. I think that the idea to create an official integration could work.

**Reply by RickCogley** [reply]
Sure thing @MatheusRV, thanks; once I have a spare moment today.

**Reply by RickCogley** [reply]
ok, so, my firm sells a cloud database system based on [dbFLEX](https://www.dbflex.net/), which is sold also as "[Teamdesk](https://www.teamdesk.net/)" by the platform maker Foresoft. The systems have an [API](https://www.teamdesk.net/help/15.6.aspx), which is also documented [here](https://www.teamdesk.net/blog/whats-new/teamdesk-rest-api/). 

It's a typical REST api, so you pass a token and can pull csv or json from views on the database. This system also has triggers, so you can use what's called a "call URL" action to push data somewhere. That's where using your documents.update endpoint would make sense. You can set it to run on a schedule, and it can push json in some format that you assemble in its editor. 

What would really be cool is, if I could use a slash command in outline to pull json or csv, and render it as a grid, based on some query I pass. You generally specify a "view" that you build and use url params to specify things like "first 10" or "sort as descending" and so on. 

Even a generic /pullcsv block command would be do-able, because I can use a webhook glue service such as https://webhook.site, to pull what I need from the db, and present it so it can be consumed by Outline.

The second point I made was about images. I have found that people just don't spend the proper amount of time making images "right" for use in a doc like this; rather they just want to quickly format and paste in, unless they've got really good awareness. I've never found it hard to resize and optimize images before use, but, I know lots of people who just want to paste and be done with it. No idea why it's so hard for people but it is. So, if Outline could offer a way to right click on an image and select some icon that means 100%, 50% or 25% width, even that would be a great improvement over nothing and a good start. In my opinion of course.
